File: Guanex.cpp
#include <iostream>
#include <list>

int main()
{
    std::list<int> cola;
    int min = 0;
    int nQ = 0;
    std::cin >> nQ;
    for (size_t j = 0; j < nQ; j++)
    {
        int instr = 0;
        int val = 0;
        std::cin >> instr;
        switch (instr)
        {
        case 1:
            /* code */
            std::cin >> val;
            cola.push_back(val);
            cola.sort();
            min = cola.front();
            break;
        case 2:
            /* code */
            if (!cola.empty())
            {
                /* code */
                min = cola.back();
                cola.pop_back();
            }
            break;

        case 3:
            /* code */
            if (cola.empty())
            {
                /* code */
                std::cout << "Empty!\n";
            }
            else
            {
                std::cout << min << "\n";
            }
            break;
        default:
            break;
        }
    }
}

File: Guanex.py
def main():
    cola=[]
    min = 0
    nQ = 0
    nQ=int(input())
    for i in range(nQ):
        instr = 0
        val = 0
        instr=int(input())
        
        if (instr==1):
            val=int(input())
            cola.append(val)
            cola.sort()
            min = cola[1]
            continue
        if(instr ==2) :
            if (0!=len(cola)):
                min = cola[len(cola)-1]
                cola.pop(len(cola)-1)
            continue

        if(instr ==3) :
            if (0==len(cola)):
                print("Empty!\n")
            else:
                print(min)
            continue

main()